docker/service.go

```diff
@@ -72,21 +71,45 @@ func (s *Service) Create(ctx context.Context, options options.Create) error {
 		return err
 	}
 
-	imageName, err := s.ensureImageExists(ctx, options.NoBuild)
-	if err != nil {
+	if err := s.ensureImageExists(ctx, options.NoBuild); err != nil {
 		return err
 	}
 
 	if len(containers) != 0 {
 		return s.eachContainer(ctx, containers, func(c *Container) error {
-			return s.recreateIfNeeded(ctx, imageName, c, options.NoRecreate, options.ForceRecreate)
+			_, err := s.recreateIfNeeded(ctx, c, options.NoRecreate, options.ForceRecreate)
+			return err
 		})
 	}
 
-	_, err = s.createOne(ctx, imageName)
+	namer, err := s.namer(ctx, 1)
+	if err != nil {
+		return err
+	}
+
+	_, err = s.createContainer(ctx, namer, "", nil, false)
 	return err
 }
 
+func (s *Service) namer(ctx context.Context, count int) (Namer, error) {
+	var namer Namer
+	var err error
+
+	if s.serviceConfig.ContainerName != "" {
+		if count > 1 {
+			logrus.Warnf(`The "%s" service is using the custom container name "%s". Docker requires each container to have a unique name. Remove the custom name to scale the service.`, s.name, s.serviceConfig.ContainerName)
+		}
+		namer = NewSingleNamer(s.serviceConfig.ContainerName)
+	} else {
+		client := s.clientFactory.Create(s)
+		namer, err = NewNamer(ctx, client, s.project.Name, s.name, false)
```

Comment:
I'm still debating on wether `Namer` should talk to the API (which it does right now) or not…